platform/source/mbed_retarget.cpp

```diff
@@ -516,8 +608,13 @@ extern "C" FILEHANDLE PREFIX(_open)(const char *name, int openflags)
     }
 #endif
     return open(name, openflags_to_posix(openflags));
+#else
+    // Everything goes to the same output
+    return 1;
```

Comment:
You should check that this tries to open stdout and return STDOUT_FILENO.

Reply:
The implementation follows https://static.docs.arm.com/dui0475/l/DUI0475L_libraries_user_guide.pdf page 79.

Why should we perform the check you are suggesting?
I changed it to used the macro `STDOUT_FILENO` instead of `1` [here](https://github.com/hugueskamba/mbed-os/compare/e4b81d4200acffc75d52769755310681ac4951a2..9ab95d568e3b1b18b96f057a9ddf6a43bbeaada9).

Reply:
> Why should we perform the check you are suggesting?

I was worried that an application might try to open a different file. But we are compiling out  `int open(const char *path, int oflag, ...);` so it should be fine.